gammapy/analysis/core.py

```diff
@@ -207,6 +207,29 @@ def get_flux_points(self):
         cols = ["e_ref", "ref_flux", "dnde", "dnde_ul", "dnde_err", "is_ul"]
         log.info("\n{}".format(self.flux_points.data.table[cols]))
 
+    def get_excess_map(self):
+        """Calculate excess map with respect to the current model."""
+        excess_settings = self.config.excess_map
+        log.info("Computing excess maps.")
+
+        if self.config.datasets.type == "1d":
+            raise ValueError("Cannot compute excess map for 1D dataset")
+
+        # Here we could possibly stack the datasets if needed.
+        if len(self.datasets)>1:
+            raise ValueError("Datasets must be stacked to compute the excess map")
+
+        energy_edges = self._make_energy_axis(excess_settings.energy_edges)
+        if energy_edges is not None:
+            energy_edges = energy_edges.edges
+
+        excess_map_estimator = ExcessMapEstimator(
+            correlation_radius=excess_settings.correlation_radius,
+            energy_edges=energy_edges,
+            **excess_settings.parameters
+        )
+        self.excess_maps = excess_map_estimator.run(self.datasets[0])
```

Comment:
I just asked myself which error is thrown if the `ExcessMapEstimator` is called with a `SpectrumDataset`. I found https://github.com/gammapy/gammapy/blob/master/gammapy/estimators/excess_map.py#L154, which is basically fine, but one could maybe improve the message a bit and say which dataset type was passed and which one is expected, just because the HLI is probably more likely to be used by beginners.

Reply:
Indeed I changed the message to explicitly state that 1D datasets are not supported.